GridTools/atlas4py
# Fix RPATH build settings and add minimal test infrastructure

Fix incorrect rpath configuration when libAtlas is not found in the system and has to be built. Additionally, add simple test infrastructure using tox and pytest, to simplify the verification of the build process across different Python versions, both in local development systems and in Github CI actions.

Main changes:
- Fix RPATH settings when calling CMake in setup.py
- Add simple tests using pytests
- Add simple tox configuration to run the tests with different python versions
- Add Tox CI workflow
- Update list of supported Python versions

Notes and ToDos:
- Verify that works on MacOS system
- Add CI test for using system-wide installations of libAtlas (right now this has been verified only manually )
